undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -363,18 +370,23 @@ private void processDeployment(final WarMetaData warMetaData, final DeploymentUn
             }
             controlPoint = builder.requires(ControlPointService.serviceName(topLevelName, UndertowExtension.SUBSYSTEM_NAME));
         }
+        ServletContainerService servletContainer = deploymentUnit.getAttachment(UndertowAttachments.SERVLET_CONTAINER_SERVICE);
         if (sharedSessionManagerConfig != null) {
             final ServiceName parentServiceName = deploymentUnit.getParent().getServiceName();
             sessionManagerFactory = builder.requires(parentServiceName.append(SharedSessionManagerConfig.SHARED_SESSION_MANAGER_SERVICE_NAME));
-            sessionIdentifierCodec = builder.requires(parentServiceName.append(SharedSessionManagerConfig.SHARED_SESSION_IDENTIFIER_CODEC_SERVICE_NAME));
+            if (servletContainer == null || servletContainer.getAffinityCookieConfig() == null) {
+                sessionIdentifierCodec = builder.requires(parentServiceName.append(SharedSessionManagerConfig.SHARED_SESSION_IDENTIFIER_CODEC_SERVICE_NAME));
+            } else {
+                affinityLocator = builder.requires(parentServiceName.append(SharedSessionManagerConfig.SHARED_SESSION_AFFINITY_LOCATOR_SERVICE_NAME));
+            }
```

Comment:
If using a shared session manager, we need to create the appropriate SessionConfigWrapper.  See comment below.

Reply:
Fixed. This is unfortunate consequence after so many rebases, should be all fixed now.